src/shared/identityUtils.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { CryptoUtils } from '@salesforce/lwc-dev-mobile-core';
+import { Config, ConfigAggregator } from '@salesforce/core';
+import configMeta, { ConfigVars } from './../configMeta.js';
+
+export class IdentityUtils {
+  public static async updateConfigWithIdentityToken(): Promise<void> {
+    const token = await this.getIdentityToken();
+    if (!token) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
+      const generatedIdentityToken = CryptoUtils.generateIdentityToken();
+      await this.writeIdentityToken(generatedIdentityToken);
+    }
+  }
```

Comment:
```suggestion
  public static async updateConfigWithIdentityToken(): Promise<string> {
    let token = await this.getIdentityToken();
    if (!token) {
      // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
      const generatedIdentityToken = CryptoUtils.generateIdentityToken();
      token = await this.writeIdentityToken(generatedIdentityToken);
    }
    return token;
  }
```
Could you return the token here? This will be the function that we'll use always to get the identity token. If it doesn't exist, it'll create one and return it.